server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleIndex.java

```diff
@@ -298,42 +294,59 @@ private static QueryBuilder termQuery(String field, String query, float boost) {
     /* Implementation of activation query */
     QProfileDto profile = query.getQProfile();
     if (query.getActivation() != null && profile != null) {
-
-      // ActiveRule Filter (profile and inheritance)
-      BoolQueryBuilder activeRuleFilter = boolQuery();
-      addTermFilter(activeRuleFilter, FIELD_ACTIVE_RULE_PROFILE_UUID, profile.getRulesProfileUuid());
-      addTermFilter(activeRuleFilter, FIELD_ACTIVE_RULE_INHERITANCE, query.getInheritance());
-      addTermFilter(activeRuleFilter, FIELD_ACTIVE_RULE_SEVERITY, query.getActiveSeverities());
-
-      // ChildQuery
-      QueryBuilder childQuery;
-      if (activeRuleFilter.hasClauses()) {
-        childQuery = activeRuleFilter;
-      } else {
-        childQuery = matchAllQuery();
-      }
+      QueryBuilder childQuery = buildActivationFilter(query, profile);
 
       if (TRUE.equals(query.getActivation())) {
         filters.put("activation",
-          QueryBuilders.hasChildQuery(INDEX_TYPE_ACTIVE_RULE.getType(),
-            childQuery));
+          JoinQueryBuilders.hasChildQuery(INDEX_TYPE_ACTIVE_RULE.getType(),
+            childQuery, ScoreMode.None));
       } else if (FALSE.equals(query.getActivation())) {
         filters.put("activation",
           boolQuery().mustNot(
-            QueryBuilders.hasChildQuery(INDEX_TYPE_ACTIVE_RULE.getType(),
-              childQuery)));
+            JoinQueryBuilders.hasChildQuery(INDEX_TYPE_ACTIVE_RULE.getType(),
+              childQuery, ScoreMode.None)));
       }
       QProfileDto compareToQProfile = query.getCompareToQProfile();
       if (compareToQProfile != null) {
         filters.put("comparison",
-          QueryBuilders.hasChildQuery(INDEX_TYPE_ACTIVE_RULE.getType(),
-            boolQuery().must(QueryBuilders.termQuery(FIELD_ACTIVE_RULE_PROFILE_UUID, compareToQProfile.getRulesProfileUuid()))));
+          JoinQueryBuilders.hasChildQuery(
+            INDEX_TYPE_ACTIVE_RULE.getType(),
+            boolQuery().must(QueryBuilders.termQuery(FIELD_ACTIVE_RULE_PROFILE_UUID, compareToQProfile.getRulesProfileUuid())),
+            ScoreMode.None));
       }
     }
 
     return filters;
   }
 
+  private static BoolQueryBuilder buildTagsFilter(RuleQuery query) {
+    BoolQueryBuilder q = boolQuery();
+    query.getTags().stream()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "getTags" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)